li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaFederatedConsumerImpl.java

```diff
@@ -372,17 +439,60 @@ public Long safeOffset(TopicPartition tp) {
 
   @Override
   public void close() {
-    throw new UnsupportedOperationException("Not implemented yet");
+    close(Integer.MAX_VALUE, TimeUnit.MILLISECONDS);
   }
 
   @Override
   public void close(long timeout, TimeUnit timeUnit) {
-    throw new UnsupportedOperationException("Not implemented yet");
+    // Get an immutable copy of the current set of consumers.
+    List<Map.Entry<ClusterDescriptor, LiKafkaConsumer<K, V>>> consumers = _consumers;
+
+    if (consumers.isEmpty()) {
+      LOG.info("no consumers to close for cluster group {}", _clusterGroup);
+      return;
+    }
+
+    LOG.info("closing LiKafkaConsumer for cluster group {} in {} {}...", _clusterGroup, timeout, timeUnit);
```

Comment:
i'd print how many consumers we have in there as well

Reply:
done